File: TaskMasterAPI/src/main/java/com/devlab/taskmaster/dto/project/ProjectDTO.java
package com.devlab.taskmaster.dto.project;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ProjectDTO {

    private Integer id;

    private String name;

    private String description;

    private List<ProjectTaskDTO> tasks;

    private ProjectUserDTO assignee;
}

File: TaskMasterWeb/TaskMasterWeb.Client/Dtos/Project/ProjectDto.cs
namespace TaskMasterWeb.Client.Dtos.Project
{
    public record ProjectDto
    {
        public int Id { get; set; }

        public string Name { get; set; } = default!;

        public string Description { get; set; } = default!;

        public IEnumerable<ProjectTaskDto>? Tasks { get; set; }

        public ProjectUserDto Assignee { get; set; } = default!;
    }
}